crates/sui-core/src/validator_client_monitor/stats.rs

```diff
@@ -127,149 +128,73 @@ impl ClientObservedStats {
-                OperationType::Effects => self.config.score_weights.effects_latency_weight,
-                OperationType::HealthCheck => self.config.score_weights.health_check_latency_weight,
-                OperationType::FastPath => self.config.score_weights.fast_path_latency_weight,
-                OperationType::Consensus => self.config.score_weights.consensus_latency_weight,
-            };
+        let Some(stats) = self.validator_stats.get(validator) else {
+            return final_latency;
+        };
 
-            if tx_type == TxType::SingleWriter && op == OperationType::Consensus {
-                continue;
+        if let Some(exclusion_time) = stats.exclusion_time {
+            if exclusion_time.elapsed() < self.config.failure_cooldown {
+                return final_latency;
             }
+        }
 
-            if tx_type == TxType::SharedObject && op == OperationType::FastPath {
-                continue;
-            }
+        let operation = match tx_type {
+            TxType::SharedObject => OperationType::Consensus,
+            TxType::SingleWriter => OperationType::FastPath,
+        };
 
-            // Skip if max latency is missing for this operation
-            let Some(max_latency) = max_latencies.get(&op) else {
-                continue;
+        // Get the latency for the relevant operation
+        if let Some(latency) = stats.average_latencies.get(&operation) {
+            let base_latency = latency.get();
+            let reliability = stats.reliability.get();
+            let reliability_weight = self.config.reliability_weight;
+
+            // Apply reliability penalty based on the configured weight
+            // reliability_weight = 0.0: no reliability consideration, use base latency
+            // reliability_weight = 1.0: full reliability penalty, divide by reliability
+            // reliability_weight in (0.0, 1.0): interpolate between base latency and reliability-penalized latency
+            let penalty_factor = if reliability > 0.0 && reliability_weight > 0.0 {
+                // Calculate the penalty factor: 1.0 (no penalty) to 1.0/reliability (full penalty)
+                let max_penalty = 1.0 / reliability;
+                1.0 + reliability_weight * (max_penalty - 1.0)
+            } else {
+                1.0
             };
 
-            // If validator has local stats, use them; otherwise assume max latency (conservative)
-            let latency = stats
-                .average_latencies
-                .get(&op)
-                .map(|ma| ma.get())
-                .unwrap_or(*max_latency);
-
-            // Lower latency ratios are better (inverted for scoring)
-            let latency_ratio = (latency / max_latency).min(1.0);
-            latency_score += (1.0 - latency_ratio) * latency_weight;
-            total_weight += latency_weight;
+            final_latency = (base_latency * penalty_factor).min(final_latency);
```

Comment:
How about this model? There is no division of 0 concerns, and I think it is intuitive enough (less reliability or higher reliability weight results in more reliability penalty).
```suggestion
            final_latency = (base_latency + (1.0 - reliability) * reliability_weight * MAX_LATENCY).min(MAX_LATENCY);
```